app/models/listing.rb

```diff
@@ -222,10 +222,13 @@ def self.find_with(params, current_user=nil, current_community=nil, per_page=100
     if params[:share_type].present?
       direction = params[:share_type]
       transaction_type_direction_map = ListingShapeHelper.transaction_types_to_direction_map(current_community) # deprecated
+
+      all_shapes = ListingService::API::Api.shapes.get(community_id: current_community.id).maybe.or_else([])
```

Comment:
These could be passed in from the controller calling this to avoid calling services inside model.